src/query/query_executor.c

```diff
@@ -15360,131 +15339,114 @@ qexec_execute_connect_by (THREAD_ENTRY * thread_p, XASL_NODE * xasl, XASL_STATE
+		   * value yet */
+		  db_make_int (isleaf_valp, isleaf_value);
+
+		  /* preserve the parent position pseudocolumn value */
+		  if (qexec_get_tuple_column_value (tuple_rec.tpl,(xasl->outptr_list->valptr_cnt - PCOL_PARENTPOS_TUPLE_OFFSET),
+						    parent_pos_valp, &tp_Bit_domain) != NO_ERROR)
 		    {
 		      GOTO_EXIT_ON_ERROR;
 		    }
 
-		  if (cycle == 0)
+		  /* make the "final" parent tuple */
+		  tuple_rec = temp_tuple_rec;
+		  if (qdata_copy_valptr_list_to_tuple (thread_p, connect_by->prior_outptr_list, &xasl_state->vd,
+						       &tuple_rec) != NO_ERROR)
 		    {
-		      isleaf_value = 0;
+		      GOTO_EXIT_ON_ERROR;
 		    }
+		  temp_tuple_rec = tuple_rec;
 
-		  /* found ISLEAF, and we already know LEVEL; we need to add the parent tuple into result list ASAP,
-		   * because we need the information about its position into the list to be kept into each child tuple */
-		  if (!parent_tuple_added)
+		  /* add parent tuple to output list file, and get its position into the list */
+		  if (qfile_add_tuple_get_pos_in_list (thread_p, listfile0, tuple_rec.tpl, &parent_pos) != NO_ERROR)
 		    {
-		      if (listfile1 == connect_by->start_with_list_id)
-			{
-			  if (!DB_IS_NULL (index_valp) && index_valp->need_clear == true)
-			    {
-			      pr_clear_value (index_valp);
-			    }
+		      GOTO_EXIT_ON_ERROR;
+		    }
 
-			  /* set index string pseudocolumn value to tuples from START WITH list */
-			  db_make_string (index_valp, father_index);
-			}
+		  /* set parent tuple position pseudocolumn value */
+		  db_make_bit (parent_pos_valp, DB_DEFAULT_PRECISION, REINTERPRET_CAST (DB_C_BIT, &parent_pos),
+			       sizeof (parent_pos) * 8);
 
-		      /* set CONNECT_BY_ISLEAF pseudocolumn value; this is only for completion, we don't know its final
-		       * value yet */
-		      db_make_int (isleaf_valp, isleaf_value);
+		  parent_tuple_added = true;
+		}
 
-		      /* preserve the parent position pseudocolumn value */
-		      if (qexec_get_tuple_column_value (tuple_rec.tpl,
-							(xasl->outptr_list->valptr_cnt - PCOL_PARENTPOS_TUPLE_OFFSET),
-							parent_pos_valp, &tp_Bit_domain) != NO_ERROR)
+	      /* only add a child if it doesn't create a cycle or if cycles should be ignored */
+	      if (cycle == 0 || XASL_IS_FLAGED (xasl, XASL_IGNORE_CYCLES))
+		{
+		  if (has_order_siblings_by)
+		    {
+		      if (qexec_insert_tuple_into_list (thread_p, listfile2_tmp, xasl->outptr_list, &xasl_state->vd,
+						        tplrec) != NO_ERROR)
 			{
 			  GOTO_EXIT_ON_ERROR;
 			}
-
-		      /* make the "final" parent tuple */
-		      tuple_rec = temp_tuple_rec;
-		      if (qdata_copy_valptr_list_to_tuple (thread_p, connect_by->prior_outptr_list, &xasl_state->vd,
-							   &tuple_rec) != NO_ERROR)
+		    }
+		  else
+		    {
+		      index++;
+		      son_index[0] = 0;
+		      if (bf2df_str_son_index (thread_p, &son_index, father_index, &len_son_index, index) != NO_ERROR)
```

Comment:
is not needed er_set ?

Reply:
This is legacy code. If there is no requirement for error code in this case, I don't think er_set() is not need to add.